src/recovery/cubic.rs

```diff
@@ -463,32 +476,32 @@ mod tests {
         let now = Instant::now();
         let prev_cwnd = r.cwnd();
 
-        // Send initcwnd full MSS packets to become no longer app limited
-        for _ in 0..recovery::INITIAL_WINDOW_PACKETS {
-            r.on_packet_sent_cc(r.max_datagram_size, now);
-        }
-
         // Trigger congestion event to update ssthresh
         r.congestion_event(now, packet::EPOCH_APPLICATION, now);
 
         // After congestion event, cwnd will be reduced.
         let cur_cwnd = (prev_cwnd as f64 * BETA_CUBIC) as usize;
         assert_eq!(r.cwnd(), cur_cwnd);
 
+        // Shift current time to 1 RTT
         let rtt = Duration::from_millis(100);
+        let now = now + rtt;
+        r.update_rtt(rtt, Duration::from_millis(0), now);
 
-        let acked = vec![Acked {
-            pkt_num: 0,
-            // To exit from recovery
-            time_sent: now + rtt,
-            size: r.max_datagram_size,
-        }];
+        // During Congestion Avoidance, it will take
+        // 3 ACKs (40x bytes per ACK) to increase cwnd by 1 MSS.
+        for _ in 0..3 {
+            let now = now + rtt;
+            let acked = vec![Acked {
```

Comment:
More of a general observation but it seems like we could save some lines (and potential copy/paste errors) across the codebase if we had an `Acked::new()` method.

Reply:
or `Copy` trait?